src/runtime/hexagon_remote/sim_remote.cpp

```diff
@@ -0,0 +1,271 @@
+                   uint8_t *closure) {
+    return f(user_context, idx, closure);
+}
+
+int halide_do_par_for(void *user_context, halide_task_t f,
+                      int min, int size, uint8_t *closure) {
+    for (int x = min; x < min + size; x++) {
+        int result = halide_do_task(user_context, f, x, closure);
+        if (result) {
+            return result;
+        }
+    }
+    return 0;
+}
+
+void *halide_get_symbol(const char *name) {
+    return dlsym(RTLD_DEFAULT, name);
+}
+
+void *halide_load_library(const char *name) {
+    return dlopen(name, RTLD_LAZY);
+}
+
+void *halide_get_library_symbol(void *lib, const char *name) {
+    return dlsym(lib, name);
+}
+
+typedef int (*set_runtime_t)(halide_malloc_t user_malloc,
+                             halide_free_t custom_free,
+                             halide_print_t print,
+                             halide_error_handler_t error_handler,
+                             halide_do_par_for_t do_par_for,
+                             halide_do_task_t do_task,
+                             void *(*)(const char *),
+                             void *(*)(const char *),
+                             void *(*)(void *, const char *));
+
+int initialize_kernels(const unsigned char *code, int codeLen,
+                       handle_t *module_ptr) {
+    const char *filename = (const char *)code;
+    void *lib = dlopen(filename, RTLD_LOCAL | RTLD_LAZY);
+    if (!lib) {
+        halide_print(NULL, "dlopen failed\n");
+        halide_print(NULL, dlerror());
+        return -1;
+    }
+
+    // Initialize the runtime. The Hexagon runtime can't call any
+    // system functions (because we can't link them), so we put all
+    // the implementations that need to do so here, and pass poiners
+    // to them in here.
+    set_runtime_t set_runtime = (set_runtime_t)dlsym(lib, "halide_noos_set_runtime");
+    if (!set_runtime) {
+        dlclose(lib);
+        halide_print(NULL, "halide_noos_set_runtime not found in shared object\n");
+        return -1;
+    }
+
+    int result = set_runtime(halide_malloc,
+                             halide_free,
+                             halide_print,
+                             halide_error,
+                             halide_do_par_for,
+                             halide_do_task,
+                             halide_get_symbol,
+                             halide_load_library,
+                             halide_get_library_symbol);
+    if (result != 0) {
+        dlclose(lib);
+        halide_print(NULL, "set_runtime failed\n");
+        return result;
+    }
+    *module_ptr = reinterpret_cast<handle_t>(lib);
+
+    return 0;
+}
+
+handle_t get_symbol(handle_t module_ptr, const char* name, int nameLen) {
+    return reinterpret_cast<handle_t>(dlsym(reinterpret_cast<void*>(module_ptr), name));
+}
+
+int run(handle_t module_ptr, handle_t function,
+        const buffer *input_buffersPtrs, int input_buffersLen,
```

Comment:
This is to match the results of the IDL compiler: https://github.com/halide/Halide/blob/hexagon2/src/runtime/hexagon_remote/bin/src/halide_hexagon_remote.h